src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1077,32 +1117,44 @@ int xcldev::device::validate(bool quick)
         return withWarning ? 1 : 0;
 
     // Perform DMA test
-    retVal = runOneTest("DMA test",
-            std::bind(&xcldev::device::dmatest, this, 0, false));
+    if (is_u50) {
+        retVal = runTestCase(std::string("xbtest"),
+                std::string("dma.json"), output);
+    } else {
+        retVal = runOneTest("DMA test",
+                std::bind(&xcldev::device::dmatest, this, 0, false));
+    }
     withWarning = withWarning || (retVal == 1);
     if (retVal < 0)
         return retVal;
 
     // Test bandwidth kernel
-    retVal = runOneTest("device memory bandwidth test",
-            std::bind(&xcldev::device::bandwidthKernelTest, this));
-    withWarning = withWarning || (retVal == 1);
-    if (retVal < 0)
-        return retVal;
-
-    // Perform P2P test
-    retVal = runOneTest("PCIE peer-to-peer test",
-            std::bind(&xcldev::device::testP2p, this));
+    if (is_u50) {
+        retVal = runTestCase(std::string("xbtest"),
+                std::string("memory.json"), output);
+    } else {
+        retVal = runOneTest("device memory bandwidth test",
+                std::bind(&xcldev::device::bandwidthKernelTest, this));
+    }
     withWarning = withWarning || (retVal == 1);
     if (retVal < 0)
         return retVal;
 
-    //Perform M2M test
-    retVal = runOneTest("memory-to-memory DMA test",
-            std::bind(&xcldev::device::testM2m, this));
-    withWarning = withWarning || (retVal == 1);
-    if (retVal < 0)
-        return retVal;
+    if (!is_u50) {
```

Comment:
why make specific code for u50 here? what happens if we run through this code on u50?